lms/djangoapps/learner_dashboard/tests/test_programs.py

```diff
@@ -261,10 +262,29 @@ def test_certificates_listed(self):
                 expected_credential['certificate_url']
             )
 
+    def test_switch_to_catalog(self):
+        """
+        Verify that the 'get_programs_from_catalog' switch can be used to route
+        traffic between the programs and catalog services.
+        """
+        self.create_programs_config()
+        switch_name = 'get_programs_from_catalog'
+
+        with mock.patch('openedx.core.djangoapps.programs.utils.get_programs') as mock_get_programs:
+            mock_get_programs.return_value = self.data
+
+            toggle_switch(switch_name)
+            self.client.get(self.url)
```

Comment:
Just in case the default value changes, should you explicitly set 'active' to True?